api.ts

```diff
@@ -1372,6 +1376,24 @@ export interface DynamicSyncTableOptions<
    * this sync table (including autocomplete formulas).
    */
   connectionRequirement?: ConnectionRequirement;
+  /**
+   * If true, when subsequent syncs discover new schema fields, these fields will not automatically be
+   * added as new columns on the table. The user can still manually add columns for these new fields.
+   * This only applies to tables that use dynamic schemas.
+   *
+   * When tables with dynamic schemas are synced, the {@link getSchema} formula is run each time,
+   * which may return a schema that is different than that from the last sync. The default behavior
+   * is that any schema fields that are new in this sync are automatically added as new columns,
+   * so they are apparent to the user. However, in rare cases when schemas change frequently,
+   * this can cause the number of columns to grow quickly and become overwhelming. Setting this
+   * value to true leaves the columns unchanged and puts the choice of what columns to display
+   * into the hands of the user.
+   */
+  doNotAddNewSyncColumnsByDefault?: boolean;
+  /**
+   * Optional placeholder schema before the dynamic schema is retrieved.
```

Comment:
What's the behavior if the user doesn't supply the placeholder schema? What's the advantage of supplying one?

Reply:
If the sync table has any parameter, we won't trigger the sync right away when it's added to the canvas. At this moment, we'll still render a placeholder table based on if there's any placeholder schema given. 

For example, Jira passes in a placeholder schema of 6 columns (issue, reporter, start date, etc) that would render in this phase. Without a placeholder schema, the temporary table would just have one column with the object name. 

In addition, if `doNotAddNewSyncColumnsByDefault`/`disableNewColumnsOnSync` is set to true, we would only add the placeholder schema columns to the canvas by default. (hmm I think I need to add this to document as well)